src/aggregator/client/m3msg_client.go

```diff
@@ -226,7 +227,8 @@ func (c *M3MsgClient) WriteForwarded(
 
 //nolint:gocritic
 func (c *M3MsgClient) write(metricID id.RawID, payload payloadUnion) error {
-	shard := c.shardFn(metricID, c.m3msg.numShards)
+	shardingId, _ := parsers.GetMetricIDWithoutLe(metricID)
```

Comment:
What's the plan for rolling back if things go wrong? Do we need a feature flag or reverting the image suffices?

Reply:
We can use a feature flag in m3-shard-flag with a new image, and we can deploy to dev region first.

Reply:
Keep it in mind that m3 Aggregator is a statefulset. Updating it is slow.

Reply:
After some more thought, the order of update doesn't really matter. This feature only changes the code path for metrics with `le` label (since they are now sent to the same shard). Other metrics will still be sent to the same shards as before, so no change in behavior there. 

Since histogram metric aggregation is not supported, nobody cares about the result right now, so we can afford some inaccuracies when restarting for the first time. What do you think @hczhu @davidyuanfs ?